jerry-core/debugger/debugger.c

```diff
@@ -586,6 +586,17 @@ jerry_debugger_process_message (uint8_t *recv_buffer_p, /**< pointer the the rec
       return true;
     }
 
+    case JERRY_DEBUGGER_CONTEXT_RESET:
+    {
+      JERRY_DEBUGGER_CHECK_PACKET_SIZE (jerry_debugger_receive_type_t);
+
+      JERRY_CONTEXT (debugger_flags) = (uint8_t) (JERRY_CONTEXT (debugger_flags) | JERRY_DEBUGGER_CONTEXT_RESET_MODE);
```

Comment:
please make this case similar to the JERRY_DEBUGGER_NO_MORE_SOURCES above. Check mode and remove JERRY_DEBUGGER_CLIENT_SOURCE_MODE flag.